Source/GTMSessionFetcher.m

```diff
@@ -2853,6 +2857,22 @@ - (void)URLSession:(NSURLSession *)session
   }];
 }
 
+#if GTM_SUPPORTS_METRICS_COLLECTION
+- (void)URLSession:(NSURLSession *)session
+              task:(NSURLSessionTask *)task
+didFinishCollectingMetrics:(NSURLSessionTaskMetrics *)metrics {
+  GTMSessionFetcherMetricsCollectionBlock metricsCollectionBlock = _metricsCollectionBlock;
```

Comment:
This should happen inside the @synchronized block.

Reply:
Done.